<!DOCTYPE html>
<!-- saved from url=(0014)about:internet -->
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=9" >

<title>Description of variables in data files used in statistical analysis</title>

<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 12px;
   margin: 8px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 { 
   font-size:2.2em; 
}

h2 { 
   font-size:1.8em; 
}

h3 { 
   font-size:1.4em; 
}

h4 { 
   font-size:1.0em; 
}

h5 { 
   font-size:0.9em; 
}

h6 { 
   font-size:0.8em; 
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre {	
   margin-top: 0;
   max-width: 95%;
   border: 1px solid #ccc;
   white-space: pre-wrap;
}

pre code {
   display: block; padding: 0.5em;
}

code.r, code.cpp {
   background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * { 
      background: transparent !important; 
      color: black !important; 
      filter:none !important; 
      -ms-filter: none !important; 
   }

   body { 
      font-size:12pt; 
      max-width:100%; 
   }
       
   a, a:visited { 
      text-decoration: underline; 
   }

   hr { 
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote { 
      padding-right: 1em; 
      page-break-inside: avoid; 
   }

   tr, img { 
      page-break-inside: avoid; 
   }

   img { 
      max-width: 100% !important; 
   }

   @page :left { 
      margin: 15mm 20mm 15mm 10mm; 
   }
     
   @page :right { 
      margin: 15mm 10mm 15mm 20mm; 
   }

   p, h2, h3 { 
      orphans: 3; widows: 3; 
   }

   h2, h3 { 
      page-break-after: avoid; 
   }
}

</style>





</head>

<body>
<h1>Description of variables in data files used in statistical analysis</h1>

<h3>merged_base_ratios.csv</h3>

<p>[1] &ldquo;&lt;!&ndash; html table generated in R 3.0.2 by xtable 1.7-1 package &ndash;&gt;\n&lt;!&ndash; Sun Mar  9 15:26:14 2014 &ndash;&gt;\n<TABLE border=1>\n<TR> <TH>  </TH> <TH> Variable </TH> <TH> Description </TH>  </TR>\n  <TR> <TD align=\"right\"> 1 </TD> <TD> REF </TD> <TD> Reference Base </TD> </TR>\n  <TR> <TD align=\"right\"> 2 </TD> <TD> POS </TD> <TD> Reference Position </TD> </TR>\n  <TR> <TD align=\"right\"> 3 </TD> <TD> DP </TD> <TD> Approximate read depth (reads with MQ=255 or with bad mates are filtered) </TD> </TR>\n  <TR> <TD align=\"right\"> 4 </TD> <TD> QUAL </TD> <TD> Quality values assigned by variant caller </TD> </TR>\n  <TR> <TD align=\"right\"> 5 </TD> <TD> MQ </TD> <TD> RMS Mapping Quality </TD> </TR>\n  <TR> <TD align=\"right\"> 6 </TD> <TD> AF </TD> <TD> Allele Frequency, for each ALT allele, in the same order as listed </TD> </TR>\n  <TR> <TD align=\"right\"> 7 </TD> <TD> AN </TD> <TD> Number of observed alleles </TD> </TR>\n  <TR> <TD align=\"right\"> 8 </TD> <TD> Dels </TD> <TD> Fraction of Reads Containing Spanning Deletions </TD> </TR>\n  <TR> <TD align=\"right\"> 9 </TD> <TD> FS </TD> <TD> Phred-scaled p-value using Fisher&#39;s exact test to detect strand bias </TD> </TR>\n  <TR> <TD align=\"right\"> 10 </TD> <TD> HaplotypeScore </TD> <TD> Consistency of the site with at most two segregating haplotypes </TD> </TR>\n  <TR> <TD align=\"right\"> 11 </TD> <TD> dataset </TD> <TD> Sequencing dataset </TD> </TR>\n  <TR> <TD align=\"right\"> 12 </TD> <TD> A </TD> <TD> Number of As at the position </TD> </TR>\n  <TR> <TD align=\"right\"> 13 </TD> <TD> C </TD> <TD> Number of Cs at the position </TD> </TR>\n  <TR> <TD align=\"right\"> 14 </TD> <TD> G </TD> <TD> Number of Gs at the position </TD> </TR>\n  <TR> <TD align=\"right\"> 15 </TD> <TD> T </TD> <TD> Number of Ts at the position </TD> </TR>\n   </TABLE>\n&rdquo;</p>

<h3>biovar_table.csv</h3>

<p>[1] &ldquo;&lt;!&ndash; html table generated in R 3.0.2 by xtable 1.7-1 package &ndash;&gt;\n&lt;!&ndash; Sun Mar  9 15:26:14 2014 &ndash;&gt;\n<TABLE border=1>\n<TR> <TH>  </TH> <TH> Variable </TH> <TH> Description </TH>  </TR>\n  <TR> <TD align=\"right\"> 1 </TD> <TD> REF </TD> <TD> Reference Base </TD> </TR>\n  <TR> <TD align=\"right\"> 2 </TD> <TD> POS </TD> <TD> Reference Position </TD> </TR>\n  <TR> <TD align=\"right\"> 3 </TD> <TD> DP </TD> <TD> Approximate read depth (reads with MQ=255 or with bad mates are filtered) </TD> </TR>\n  <TR> <TD align=\"right\"> 4 </TD> <TD> QUAL </TD> <TD> Quality values assigned by variant caller </TD> </TR>\n  <TR> <TD align=\"right\"> 5 </TD> <TD> MQ </TD> <TD> RMS Mapping Quality </TD> </TR>\n  <TR> <TD align=\"right\"> 6 </TD> <TD> AF </TD> <TD> Allele Frequency, for each ALT allele, in the same order as listed </TD> </TR>\n  <TR> <TD align=\"right\"> 7 </TD> <TD> AN </TD> <TD> Number of observed alleles </TD> </TR>\n  <TR> <TD align=\"right\"> 8 </TD> <TD> Dels </TD> <TD> Fraction of Reads Containing Spanning Deletions </TD> </TR>\n  <TR> <TD align=\"right\"> 9 </TD> <TD> FS </TD> <TD> Phred-scaled p-value using Fisher&#39;s exact test to detect strand bias </TD> </TR>\n  <TR> <TD align=\"right\"> 10 </TD> <TD> HaplotypeScore </TD> <TD> Consistency of the site with at most two segregating haplotypes </TD> </TR>\n  <TR> <TD align=\"right\"> 11 </TD> <TD> dataset </TD> <TD> Sequencing dataset </TD> </TR>\n  <TR> <TD align=\"right\"> 12 </TD> <TD> A </TD> <TD> Number of As at the position </TD> </TR>\n  <TR> <TD align=\"right\"> 13 </TD> <TD> C </TD> <TD> Number of Cs at the position </TD> </TR>\n  <TR> <TD align=\"right\"> 14 </TD> <TD> G </TD> <TD> Number of Gs at the position </TD> </TR>\n  <TR> <TD align=\"right\"> 15 </TD> <TD> T </TD> <TD> Number of Ts at the position </TD> </TR>\n  <TR> <TD align=\"right\"> 16 </TD> <TD> prop </TD> <TD> Major base proportion (Y/N) </TD> </TR>\n  <TR> <TD align=\"right\"> 17 </TD> <TD> max_copy </TD> <TD> Major base gene copy number with the maximum probablility </TD> </TR>\n  <TR> <TD align=\"right\"> 18 </TD> <TD> max_prob </TD> <TD> Probability of max_copy </TD> </TR>\n  <TR> <TD align=\"right\"> 19 </TD> <TD> N </TD> <TD> Number of major and minor bases </TD> </TR>\n  <TR> <TD align=\"right\"> 20 </TD> <TD> Y </TD> <TD> Number of major bases </TD> </TR>\n  <TR> <TD align=\"right\"> 21 </TD> <TD> exp_ratio </TD> <TD> Consensus base ratio </TD> </TR>\n  <TR> <TD align=\"right\"> 22 </TD> <TD> total_copy </TD> <TD> Number of 16S rRNA gene copies in the genome </TD> </TR>\n  <TR> <TD align=\"right\"> 23 </TD> <TD> min_copy </TD> <TD> Minor base gene copy number, for max_copy </TD> </TR>\n  <TR> <TD align=\"right\"> 24 </TD> <TD> base_ratio </TD> <TD> Base copy ratio with the highest probability for the dataset </TD> </TR>\n   </TABLE>\n&rdquo;</p>

<h3>string_stats[_ecoli | lmono].csv</h3>

<p>[1] &ldquo;&lt;!&ndash; html table generated in R 3.0.2 by xtable 1.7-1 package &ndash;&gt;\n&lt;!&ndash; Sun Mar  9 15:26:14 2014 &ndash;&gt;\n<TABLE border=1>\n<TR> <TH>  </TH> <TH> Variable </TH> <TH> Description </TH>  </TR>\n  <TR> <TD align=\"right\"> 1 </TD> <TD> gc.1 </TD> <TD> Estimated variant combination for gene copy 1 </TD> </TR>\n  <TR> <TD align=\"right\"> 2 </TD> <TD> gc.2 </TD> <TD> Estimated variant combination for gene copy 2 </TD> </TR>\n  <TR> <TD align=\"right\"> 3 </TD> <TD> gc.3 </TD> <TD> Estimated variant combination for gene copy 3 </TD> </TR>\n  <TR> <TD align=\"right\"> 4 </TD> <TD> gc.4 </TD> <TD> Estimated variant combination for gene copy 4 </TD> </TR>\n  <TR> <TD align=\"right\"> 5 </TD> <TD> gc.5 </TD> <TD> Estimated variant combination for gene copy 5 </TD> </TR>\n  <TR> <TD align=\"right\"> 6 </TD> <TD> gc.6 </TD> <TD> Estimated variant combination for gene copy 6 </TD> </TR>\n  <TR> <TD align=\"right\"> 7 </TD> <TD> gc.7 </TD> <TD> Estimated variant combination for gene copy 7, L. monocytogenes only has 6 gene copies </TD> </TR>\n  <TR> <TD align=\"right\"> 8 </TD> <TD> likelihood </TD> <TD> Likelihood for set of variant combination </TD> </TR>\n  <TR> <TD align=\"right\"> 9 </TD> <TD> chimera </TD> <TD> Probability of a chimera event </TD> </TR>\n  <TR> <TD align=\"right\"> 10 </TD> <TD> dataset </TD> <TD> Dataset used to estimate the set of variant combinations </TD> </TR>\n   </TABLE>\n&rdquo;</p>

<h3>string_tables[_filtered].csv</h3>

<p>[1] &ldquo;&lt;!&ndash; html table generated in R 3.0.2 by xtable 1.7-1 package &ndash;&gt;\n&lt;!&ndash; Sun Mar  9 15:26:14 2014 &ndash;&gt;\n<TABLE border=1>\n<TR> <TH>  </TH> <TH> Variable </TH> <TH> Description </TH>  </TR>\n  <TR> <TD align=\"right\"> 1 </TD> <TD> variant_string </TD> <TD> Combinations of bases at biologically variable positions </TD> </TR>\n  <TR> <TD align=\"right\"> 2 </TD> <TD> counts </TD> <TD> Number of reads with the combinations of variants </TD> </TR>\n  <TR> <TD align=\"right\"> 3 </TD> <TD> dataset </TD> <TD> source sequence dataset </TD> </TR>\n   </TABLE>\n&rdquo;</p>

</body>

</html>

